p2p/host/autorelay/autorelay_test.go

```diff
@@ -294,8 +296,122 @@ func TestConnectOnDisconnect(t *testing.T) {
 	require.NotEqualf(t, oldRelay, relaysInUse[0], "old relay should not be used again")
 }
 
+type mockClock struct {
+	mu     sync.Mutex
+	now    time.Time
+	timers []*mockInstantTimer
+}
+
+var _ autorelay.ClockWithInstantTimer = &mockClock{}
+
+type mockInstantTimer struct {
```

Comment:
Move this to a separate file? We might need this at other places as well, and that makes it easier to expose later without losing Git history.